frontend/src/hooks/useCallbackAppSelector.ts

```diff
@@ -15,11 +14,13 @@ export function useCallbackAppSelector<TState = RootState, TSelected = unknown>(
   const selected: TSelected = useAppSelector(selector as any);
   const isInitialMount = useRef(true);
 
-  useDeepCompareEffect(() => {
+  useEffect(() => {
     if (isInitialMount.current) {
       isInitialMount.current = false;
     } else {
       callbackFn && callbackFn(selected);
     }
-  }, [callbackFn, selected]);
+    // TODO: currently useApi is causing this to refresh when the jwt token expires, that hook needs to be re-written
```

Comment:
the reason for this requirement is stated above. the functions returned by useApi will trigger useEffect hooks as they are regenerated whenever the axios instance is updated.

Reply:
I'm very curious about this, as I was seeing this behaviour when it was created.  But I couldn't get my head around moving this down the pipe so that it only impacts relevant components.  How is it that you have solved it?

Reply:
I've only partially solved it, see my comments on useApi for more details.

Reply:
see this comment: https://github.com/bcgov/PSP/pull/90#discussion_r657514359